operate/services/manage.py

```diff
@@ -2066,6 +2078,70 @@ def refill_requirements(  # pylint: disable=too-many-locals
             "allow_start_agent": allow_start_agent,
         }
 
+    def _compute_olas_requirements(  # pylint: disable=too-many-locals
+        self, service_config_id: str, chain: str
+    ) -> int:
+        service = self.load(service_config_id=service_config_id)
+        chain_config = service.chain_configs[chain]
+        ledger_config = chain_config.ledger_config
+        user_params = chain_config.chain_data.user_params
+
+        # TODO fix this
+        os.environ["CUSTOM_CHAIN_RPC"] = ledger_config.rpc
+
+        if not user_params.use_staking:
+            return 0
+
+        # First, compute required OLAS
+        sftxb = self.get_eth_safe_tx_builder(ledger_config=ledger_config)
+        staking_params = sftxb.get_staking_params(
+            staking_contract=STAKING[ledger_config.chain][
+                user_params.staking_program_id
+            ],
+        )
+
+        required_olas = (
+            staking_params["min_staking_deposit"]
+            + staking_params["min_staking_deposit"]  # bond = staking
+        )
+
+        # Second, compute available OLAS in wallet + in current staking contract
+        wallet = self.wallet_manager.load(ledger_config.chain.ledger_type)
+        ledger_api = wallet.ledger_api(chain=ledger_config.chain, rpc=ledger_config.rpc)
+        master_safe = wallet.safes[Chain(chain)]
+        available_olas = get_asset_balance(
+            ledger_api=ledger_api,
+            asset_address=OLAS[Chain(chain)],
+            address=master_safe,
+        )
+
+        service_id = chain_config.chain_data.token
+        if service_id == NON_EXISTENT_TOKEN:
+            return max(required_olas - available_olas, 0)
+
+        service_owner = registry_contracts.service_registry.get_service_owner(
+            ledger_api=ledger_api,
+            contract_address=staking_params["service_registry"],
+            service_id=service_id,
+        ).get("service_owner", "")
+
+        if service_owner.lower() == master_safe.lower():
+            return max(required_olas - available_olas, 0)
+
+        # If the service owner is not the master safe, assume it's staked
+        # and get current bond on the contract. Assume current security
+        # deposit = current agent bond
+        # TODO this only works for 1 agent
+        current_agent_bond = sftxb.get_agent_bond(
+            service_id=service_id, agent_id=staking_params["agent_ids"][0]
+        )
+        current_security_deposit = current_agent_bond
```

Comment:
@kupermind Is this OK? However, if there are multiple agents in a service, how to correctly compute the current security deposit ?

Reply:
If it's for the staking contract, then each bond is equal to the security deposit as well. Thus, the total bond is:
`security deposit * (1 + num_agent_instances)`

Reply:
Ok, but how to know exactly how many agents have their balance staked already. For example if I'm in the situation where I have 4 agents but i have just registered 2, how do I retrieve the value:

(1+2) * security_deposit

?

Reply:
getService(serviceId) => get the 5th field (numAgentInstances) and compare with the 4th (maxNumAgentInstances)

`maxNumAgentInstances - numAgentInstances` gives the number of free slots